### PR TITLE
ms5611: fix sleep in work process will block WorkQueue schedule.

### LINKED ISSUE #74: FIFO buffer corruption
Every now and then (about every 20th try) I get the FIFO buffer corruption on startup of the MPU9250 driver. A FIFO buffer reset of the sensor doesn't seem to fix the problem. Maybe a general sensor reset would help.

### PR DESCRIPTION
**Pull request by iZhangHui**
It will seriously reduce other sensors's collect rate, and cause issue like FIFO overflow on MPU9250 and LSM9DS1.

It was very easily reproduce on rpi3+navio2, just start ms6511 and mpu9250 drivers.

This should be fix #74 .

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Thanks! That's indeed a pretty big fail.

File: drivers/ms5611/MS5611.cpp [hunk 1]
```diff
@@ -329,66 +335,78 @@ int MS5611::_collect(uint32_t *raw)
...
-	// Request to convert the pressure
-	if (_request(ADDR_CMD_CONVERT_D1) < 0) {
-		DF_LOG_ERR("error: pressure measure failed");
-		return;
-	}
+			/* collect fails, re-initiate a pressure read command
+			 * or we are stuck.
+			 */
+			if (_request(ADDR_CMD_CONVERT_D1) < 0) {
+				DF_LOG_ERR("error: pressure measure failed");
```
Comment (julianoes) [inline drivers/ms5611/MS5611.cpp]: Shouldn't there be a `_measure_phase = 0;` in this return case?

**Comment by julianoes** [comment]
Nice, thanks @iZhangHui. Can you comment my question above?

File: drivers/ms5611/MS5611.cpp (hunk 1 above)
Comment (iZhangHui) [inline drivers/ms5611/MS5611.cpp]: This return case should move out of  `if (_request(ADDR_CMD_CONVERT_D1) < 0) { }`.
When pressure collecting failed, do a reset and re-issue a request to collect again.
The logic is same with temperature collection above.

**Comment by iZhangHui** [comment]
@julianoes Thank you for pointing out this. 

Another problem is that I forgot to initialize  `_measure_phase`. I will create a new PR to fix them.

**Pull request merged**